lib/prompt/parent_branches.go

```diff
@@ -33,5 +46,70 @@ func askForBranchAncestry(branchName string) {
 }
 
 func askForParentBranch(branchName string) string {
-	panic("unimplemented")
+	branchNameFmt := color.New(color.Bold).Add(color.FgCyan)
+	mainBranch := config.GetMainBranch()
+	message := fmt.Sprintf("Please specify the parent branch of %s by name or number (default: %s): ", branchNameFmt.Sprintf(branchName), mainBranch)
+	numberedBranches := git.GetLocalBranchesWithMainBranchFirst()
+	numericRegex, err := regexp.Compile("^[0-9]+$")
+	if err != nil {
+		log.Fatal("Error compiling numeric regular expression: ", err)
+	}
+
+	for {
+		fmt.Printf(message)
+		userInput := util.GetUserInput()
+		parent := ""
+		if numericRegex.MatchString(userInput) {
+			index, err := strconv.Atoi(userInput)
+			if err != nil {
+				log.Fatal("Error parsing string to integer: ", err)
+			}
+			if index >= 1 && index <= len(numberedBranches) {
+				parent = numberedBranches[index-1]
+			} else {
+				util.PrintError("Invalid branch number")
+			}
+		} else if userInput == "" {
+			parent = mainBranch
+		} else if git.HasBranch(userInput) {
+			parent = userInput
+		} else {
+			util.PrintError(fmt.Sprintf("Branch '%s' doesn't exist", userInput))
+		}
+
+		if parent == "" {
+			continue
+		} else if branchName == parent {
+			util.PrintError(fmt.Sprintf("'%s' cannot be the parent of itself", parent))
+		} else if config.HasAncestorBranch(parent, branchName) {
+			util.PrintError(fmt.Sprintf("Nested branch loop detected: '%s' is an ancestor of '%s'", branchName, parent))
+		} else {
+			return parent
+		}
+	}
+}
+
+var parentBranchHeaderShown = false
+
+func printParentBranchHeader() {
+	if !parentBranchHeaderShown {
+		parentBranchHeaderShown = true
+		fmt.Println()
```

Comment:
Maybe use a multi-line string here? It would look something like this:

```go
fmt.Println(`
Feature branches can be branched directly off 
%s or from other feature branches.

The former allows to develop and ship features completely independent of each other.
The latter allows to build on top of currently unshipped features.

`, config.GetMainBranch())
printNumberedBranches()
fmt.Println()